README.md

```diff
@@ -26,22 +26,52 @@ A set of **clustergroupupgrade** example CRs can be found in the **samples** fol
 
 ## How it works
 
-The ClusterGroupUpgrade controller is designed to perform an upgrade to a group of clusters in a given period of time, which by default is set to 4h.
-In order to model the various phases of an upgrade, the ClusterGroupUpgrade CR can be in the following states:
-
-* **UpgradeNotStarted**
+The ClusterGroupUpgrade controller is designed to perform an upgrade to a group of clusters in a given period of time, which by default is set to 4h. In order to model the various phases of an upgrade, the ClusterGroupUpgrade CR can be in a variety of states.
+
+The states generally are represented by Conditions using appropriate reasons and types. The general workflow is:
+
+ClusterSelected -> BackupSucceeded -> PrecacheSpecValid -> PrecachingSucceeded -> Validated -> Progressing -> Succeeded
```

Comment:
In the next PR I'm going to move Validated to be after ClusterSelected as proposed in the original enhancement, but for now I want the doc to represent the current behaviour in case there are issues getting that work done